src/Microsoft.Restier.Providers.EntityFramework/Submit/ChangeSetInitializer.cs

```diff
@@ -40,6 +40,13 @@ internal class ChangeSetInitializer : IChangeSetInitializer
             {
                 object strongTypedDbSet = dbContext.GetType().GetProperty(entry.EntitySetName).GetValue(dbContext);
                 Type entityType = strongTypedDbSet.GetType().GetGenericArguments()[0];
+
+                // This means request entity is sub type of entity type
+                if (entry.ActualEntityType != null && entityType != entry.ActualEntityType)
```

Comment:
could `ActualEntityType` be null? if not, we should add the enforcement somewhere.